examples/systemcalls/systemcalls.c

```diff
@@ -92,8 +146,59 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    int ret;
+    pid_t pid = fork();
```

Comment:
@japa7016 think about how you could avoid duplicating logic with the function above.